thriftbp/middlewares.go

```diff
@@ -0,0 +1,81 @@
+package thriftbp
+
+import (
+	"context"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/reddit/baseplate.go/log"
+)
+
+// BaseplateProcessor is a TProcessor that can be thriftbp.Wrap-ed and
+// thriftbp.Merge-d.
+//
+// The TProcessors generated by the Apache Thrift compiler fufill this
+// interface, but not all of them are a part of any interface within Apache
+// Thrift.
+type BaseplateProcessor interface {
+	// Process fufills the TProcessor interface.
+	Process(ctx context.Context, in, out thrift.TProtocol) (bool, thrift.TException)
```

Comment:
🔕 For this one you can just embed `thrift.TProcessor`. That also makes it more obvious that anything implementing this interface also implements `thrift.TProcessor`.